spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -157,7 +158,11 @@ private void assertEquals(String context, Object[] expectedRow, Object[] actualR
       Object actualValue = actualRow[col];
       if (expectedValue != null && expectedValue.getClass().isArray()) {
         String newContext = String.format("%s (nested col %d)", context, col + 1);
-        assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
+        if (expectedValue.getClass().getComponentType().isPrimitive()) {
+          new ExactComparisonCriteria().arrayEquals(newContext, expectedValue, actualValue);
+        } else {
+          assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
+        }
```

Comment:
Why was this needed?

Reply:
Thanks for your review, I have modified the code